File: frontend/update_navbar.js
const fs = require('fs');

// Read the backup file
const content = fs.readFileSync('src/components/Navbar.tsx.backup', 'utf8');

// Split into lines
const lines = content.split('\n');

// Find and update desktop navigation (around line 52-56)
let desktopUpdated = false;
let mobileUpdated = false;

for (let i = 0; i < lines.length; i++) {
  // Desktop navigation update
  if (!desktopUpdated && lines[i].includes("{ href: '/subscripcions', label: 'Planes' },") && 
      i > 50 && i < 60 && lines[i+1].includes("{ href: '/contact', label: 'Contacto' }")) {
    lines.splice(i + 1, 0, "                ...(isAuthenticated ? [{ href: '/videos', label: 'Videos' }] : []),");
    desktopUpdated = true;
    i++; // Skip the line we just added
  }
  
  // Mobile navigation update  
  if (!mobileUpdated && lines[i].includes("{ href: '/subscripcions', label: 'Planes' },") && 
      i > 180 && i < 200 && lines[i+1].includes("{ href: '/contact', label: 'Contacto' }")) {
    lines.splice(i + 1, 0, "                ...(isAuthenticated ? [{ href: '/videos', label: 'Videos' }] : []),");
    mobileUpdated = true;
  }
}

// Write the modified content
fs.writeFileSync('src/components/Navbar.tsx', lines.join('\n'), 'utf8');

console.log('Navbar.tsx updated successfully!');
console.log('Desktop navigation updated:', desktopUpdated);
console.log('Mobile navigation updated:', mobileUpdated);

File: frontend/update_navbar.py
import re

# Read the backup file
with open('src/components/Navbar.tsx.backup', 'r', encoding='utf-8') as f:
    content = f.read()

# First replacement - desktop navigation (around line 52-56)
desktop_pattern = r"(\{\/\* Centered Navigation \(desktop\) \*\/\}.*?)\{\[\s*\{ href: '/', label: 'Inicio' \},\s*\{ href: '/subscripcions', label: 'Planes' \},\s*\{ href: '/contact', label: 'Contacto' \}\s*\]\.map\(\(item\) => \{"
desktop_replacement = r"\1{[\n                { href: '/', label: 'Inicio' },\n                { href: '/subscripcions', label: 'Planes' },\n                ...(isAuthenticated ? [{ href: '/videos', label: 'Videos' }] : []),\n                { href: '/contact', label: 'Contacto' }\n              ].map((item) => {"

content = re.sub(desktop_pattern, desktop_replacement, content, flags=re.DOTALL)

# Second replacement - mobile navigation (around line 184-188)
mobile_pattern = r"(\{\/\* Mobile Navigation \*\/\}.*?)\{\[\s*\{ href: '/', label: 'Inicio' \},\s*\{ href: '/subscripcions', label: 'Planes' \},\s*\{ href: '/contact', label: 'Contacto' \}\s*\]\.map\(\(item\) => \{"
mobile_replacement = r"\1{[\n                { href: '/', label: 'Inicio' },\n                { href: '/subscripcions', label: 'Planes' },\n                ...(isAuthenticated ? [{ href: '/videos', label: 'Videos' }] : []),\n                { href: '/contact', label: 'Contacto' }\n              ].map((item) => {"

content = re.sub(mobile_pattern, mobile_replacement, content, flags=re.DOTALL)

# Write the modified content
with open('src/components/Navbar.tsx', 'w', encoding='utf-8') as f:
    f.write(content)

print("Navbar.tsx updated successfully!")
